### PR TITLE
feat(client) ts-migrate LearnLayout Component

### PR DESCRIPTION
**Pull request by sharmaabhinav**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/layouts/learn.tsx [hunk 1]
```diff
@@ -35,7 +44,17 @@ const mapDispatchToProps = {
 
 const RedirectEmailSignUp = createRedirect('/email-sign-up');
 
-class LearnLayout extends Component {
+type LearnLayoutProps = {
+  isSignedIn: boolean;
+  fetchState: fetchState;
+  user: user;
+  tryToShowDonationModal: () => void;
+  children: React.ReactNode;
```
Comment (thecodingaviator) [inline client/src/components/layouts/learn.tsx:52]: Try to leave optional props as optional

Comment (thecodingaviator) [inline client/src/components/layouts/learn.tsx:52]: Try to leave optional props as optional

**Review by thecodingaviator — DISMISSED** [review]

**Review by thecodingaviator — COMMENTED** [review]
The optional props issue looks good now. Fix the lining and this should be good to go

File: client/src/components/layouts/learn.tsx [hunk 2]
```diff
@@ -18,11 +17,21 @@ import './prism-night.css';
 import 'react-reflex/styles.css';
 import './learn.css';
 
+type fetchState = {
```
Comment (ShaunSHamilton) [inline client/src/components/layouts/learn.tsx:20]:
```suggestion
type FetchState = {
```

File: client/src/components/layouts/learn.tsx [hunk 3]
```diff
@@ -18,11 +17,21 @@ import './prism-night.css';
...
+  pending: boolean;
+  complete: boolean;
+  errored: boolean;
+};
+
+type user = {
```
Comment (ShaunSHamilton) [inline client/src/components/layouts/learn.tsx:26]:
```suggestion
type User = {
```

File: client/src/components/layouts/learn.tsx [hunk 4]
```diff
@@ -18,11 +17,21 @@ import './prism-night.css';
...
+  acceptedPrivacyTerms: boolean;
+};
+
 const mapStateToProps = createSelector(
   userFetchStateSelector,
   isSignedInSelector,
   userSelector,
-  (fetchState, isSignedIn, user) => ({
+  (fetchState: fetchState, isSignedIn, user: user) => ({
```
Comment (ShaunSHamilton) [inline client/src/components/layouts/learn.tsx:34]:
```suggestion
  (fetchState: FetchState, isSignedIn, user: User) => ({
```

File: client/src/components/layouts/learn.tsx [hunk 5]
```diff
@@ -35,7 +44,17 @@ const mapDispatchToProps = {
 
 const RedirectEmailSignUp = createRedirect('/email-sign-up');
 
-class LearnLayout extends Component {
+type LearnLayoutProps = {
+  isSignedIn?: boolean;
+  fetchState: fetchState;
+  user: user;
```
Comment (ShaunSHamilton) [inline client/src/components/layouts/learn.tsx:50]:
```suggestion
  fetchState: FetchState;
  user: User;
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Just nitpicks for consistency

File: client/src/components/layouts/learn.tsx [hunk 6]
```diff
@@ -88,6 +88,8 @@ class LearnLayout extends Component<LearnLayoutProps> {
           <meta content='noindex' name='robots' />
         </Helmet>
         <main id='learn-app-wrapper'>{children}</main>
+        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */
+        /* @ts-ignore  */}
```
Comment (sharmaabhinav) [inline client/src/components/layouts/learn.tsx:92]: Migration of `DonationModal.tsx` is required to fix hence added this a last option as per guidelines

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Comment by ShaunSHamilton** [comment]
@thecodingaviator Feel free to merge, if you are happy 🙂

**Review by thecodingaviator — COMMENTED** [review]
LGTM!

**Pull request merged**